src/core/internal/convert.d

```diff
@@ -10,13 +10,33 @@
 module core.internal.convert;
 import core.internal.traits : Unqual;
 
-@trusted pure nothrow
+/+
+A @nogc function can allocate memory during CTFE.
++/
+@nogc nothrow pure @trusted
+private ubyte[] ctfe_alloc()(size_t n)
+{
+    if (!__ctfe)
+    {
+        assert(0, "CTFE only");
+    }
+    else
+    {
+        static ubyte[] alloc(size_t x)
+        {
+            return new ubyte[x];
+        }
```

Comment:
So it's clear we want to add `@nogc` to the allocation artificially. But since you don't have the `pure` and `nothrow` attributes on here, it's not required that they are inferred pure/nothrow. But it's definitely possible for someone to come along and put in impure or throwing functions in here, and all of a sudden you have casted that away below.

2 options I see: a) attribute the function (can that even be done? I can't remember), b) static assert the type of the function before casting.

Reply:
Added attributes.